communication/nats/sender.go

```diff
@@ -94,3 +95,12 @@ func (sender *senderNATS) Request(producer communication.RequestProducer) (respo
 
 	return responsePtr, nil
 }
+
+func checkConnection(conn Connection) {
```

Comment:
I would like to see this function part of connection. I.e. conn.Check

Reply:
Changed it to be part of the connection.